### PR TITLE
Add admin auth middleware

### PR DESCRIPTION
**Pull request by nanu-c**
# Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# Description

This introduces an auth admin + vendor middleware. This needs a re-import to keycloak because of the missing openid scope that isn't standard in keycloak 20 anymore. That means the userinfo endpoint doesn't work.

Bonus: enable the linter #33 
# Checklist:

- [x] I have commented my code (or ChatGPT did), particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings, neither in my IDE nor in my browser
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: app/handlers/handlers_test.go [hunk 1]
```diff
@@ -2,26 +2,57 @@ package handlers
...
 	"testing"
 	"time"
 
+	"github.com/Nerzal/gocloak/v13"
 	"github.com/go-chi/chi/v5"
 	"github.com/stretchr/testify/require"
 )
 
 var r *chi.Mux
+var addminUser string
```
Comment (lebe1) [inline app/handlers/handlers_test.go:22]: This looks like a small typo to me with double 'd'

**Review by lebe1 — COMMENTED** [review]
Everything looks good except for a little typo.
Only thing not working for me locally is testing it in my own docker container.
Either I get ` /api/vendors/ expected status code 200, got 401: Unauthorized` or `Error logging in Keycloak admin could not get token: Post "http://localhost:8080/realms/master/protocol/openid-connect/token": dial tcp 127.0.0.1:8080: connect: connection refused`
Screenshot from 2023-09-19 16-20-40

**Comment by nanu-c** [comment]
> Everything looks good except for a little typo. Only thing not working for me locally is testing it in my own docker container. Either I get ` /api/vendors/ expected status code 200, got 401: Unauthorized` or `Error logging in Keycloak admin could not get token: Post "http://localhost:8080/realms/master/protocol/openid-connect/token": dial tcp 127.0.0.1:8080: connect: connection refused` Screenshot from 2023-09-19 16-20-40

the keycloak url in the .env must be `http://keycloak:8080`

File: app/handlers/handlers_test.go (hunk 1 above)
Comment (nanu-c) [reply app/handlers/handlers_test.go]: You are right this shouldn't be double d.

**Review by lebe1 — DISMISSED** [review]
Sorry for pushing so many new branches into main. Can you update your branch with main and then I will have to review again anyways?

**Review by lebe1 — APPROVED** [review]

**Pull request merged**